java/code/src/com/redhat/rhn/manager/formula/FormulaManager.java

```diff
@@ -342,4 +342,48 @@ public void enableFormula(String minionId, String formulaName) throws IOExceptio
             FormulaFactory.saveServerFormulas(minionId, enabledFormulas);
         }
     }
+
+    /**
+     * Configure system for monitoring when the entitlement is added.
+     * @param minion the MinionServer to be configured for monitoring
+     */
+    public void enableMonitoringOnEntitlementAdd(MinionServer minion) {
+        try {
+            // Assign the monitoring formula to the system unless it belongs to a group having monitoring enabled
+            if (!FormulaFactory.isMemberOfGroupHavingMonitoring(minion)) {
+                List<String> formulas = FormulaFactory.getFormulasByMinionId(minion.getMinionId());
+                if (!formulas.contains(FormulaFactory.PROMETHEUS_EXPORTERS)) {
+                    formulas.add(FormulaFactory.PROMETHEUS_EXPORTERS);
+                    FormulaFactory.saveServerFormulas(minion.getMinionId(), formulas);
+                }
+            }
+        }
+        catch (UnsupportedOperationException | IOException e) {
+            // FIXME: error handling
+            // LOG.error("Error assigning formula: " + e.getMessage(), e);
+            // result.addError(new ValidatorError("system.entitle.formula_error"));
+        }
+    }
+
+    /**
+     * Configure the monitoring formula for cleanup (disable exporters) if needed.
+     * @param minion the MinionServer to be configured for monitoring cleanup
+     */
+    public void disableMonitoringOnEntitlementRemoval(MinionServer minion) {
+        if (this.isMonitoringCleanupNeeded(minion)) {
+            try {
+                // Get the current data and set all exporters to disabled
+                String minionId = minion.getMinionId();
+                Map<String, Object> data = FormulaFactory
+                        .getFormulaValuesByNameAndMinionId(PROMETHEUS_EXPORTERS, minionId)
+                        .orElse(FormulaFactory.getPillarExample(PROMETHEUS_EXPORTERS));
+                FormulaFactory.saveServerFormulaData(
+                        FormulaFactory.disableMonitoring(data), minionId, PROMETHEUS_EXPORTERS);
+            }
+            catch (UnsupportedOperationException | IOException e) {
+                // FIXME: error handling
+                // LOG.warn("Exception on saving formula data: " + e.getMessage());
```

Comment:
i think the logging is fine here until we maybe tackle more consistent error handling overall.